client/src/components/Block.tsx

```diff
@@ -1,49 +1,101 @@
 import React, { useContext } from "react";
 import "../styling/Maze.css";
 import { PageContext } from "../PageProvider";
+import { SetState, Maze, Point } from "../utilities/types";
+import { updateMaze } from "../utilities/utilities";
 
 type BlockProps = {
   blockType: number;
   mazeSize: number;
+  maze: Maze;
+  rowIndex: number;
+  colIndex: number;
+  setMaze: SetState<Maze>;
 };
 
-const Block: React.FC<BlockProps> = ({ blockType, mazeSize }) => {
+const Block: React.FC<BlockProps> = ({ blockType, mazeSize, maze, rowIndex, colIndex, setMaze }) => {
   const {currentPage} = useContext(PageContext)
+  const cell: Point = {x: rowIndex, y: colIndex}
 
   const maze_block = {
     width: `${mazeSize}%`,
     aspectRatio: "1",
     border: "1px solid #ccc",
   };
 
+  const blockTypeColor = (blockType: number) => {
+    switch (blockType) {
+      case 0: return "maze__path"
+      case 1: return "maze__wall"
+      case 2: return "maze__walked"
+      case 3: return "maze__queued"
+      case 4: return "maze__shortpath"
+      case 5: return "maze__startpoint"
+      case 6: return "maze__endpoint"
+      default: return "";
+    }
+  }
+
+  const handleClick = (e: React.MouseEvent<HTMLDivElement>) => {
+    console.log(cell);
+    console.log(blockType);
+    const newMaze: Maze = maze
+
+    //*Set start point at top row or most left column
+    if (cell.x === 0 && cell.y !== maze.length - 1 && cell.y !== 0 || cell.x !== 0 && cell.x !== maze.length - 1 && cell.y === 0 ) {
```

Comment:
put it into a function named sth like isValidStartPoint()

Reply:
abstracted as function in utilities file